core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1094,11 +1111,11 @@ public IndexShardCacheEntity(IndexShard indexShard, QueryPhase queryPhase, Searc
              * results.*/
             final int expectedSizeInBytes = 512;
```

Comment:
I think it depends on the type of the field. For number/date fields the bounds are written as longs so 512 is maybe bigger than it should ?